static/app/views/automations/components/ruleNode.tsx

```diff
@@ -0,0 +1,75 @@
+import styled from '@emotion/styled';
+
+import {Flex} from 'sentry/components/container/flex';
+import {Button} from 'sentry/components/core/button';
+import {IconDelete} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import type {DataCondition} from 'sentry/types/workflowEngine/dataConditions';
+import {dataConditionNodesMap} from 'sentry/views/automations/components/dataConditionNodes';
+
+interface RuleNodeProps {
+  condition: Omit<DataCondition, 'condition_group' | 'type' | 'id'>;
+  condition_id: string;
+  onDelete: () => void;
+  onUpdate: (comparison: Record<string, any>) => void;
+}
+
+export default function RuleNode({
+  condition,
+  condition_id,
+  onDelete,
+  onUpdate,
+}: RuleNodeProps) {
+  return (
+    <RuleRowContainer>
+      <RuleRow>
+        <Rule>
+          {(() => {
+            const node = dataConditionNodesMap[condition.comparison_type];
+            const configNode = node?.configNode;
+            if (configNode) {
+              return configNode(condition, condition_id, onUpdate);
+            }
+            return node?.label;
+          })()}
```

Comment:
This inline IIFE feels odd. Maybe this logic should be extracted to its own component?